src/BrowserFetcher.re

```diff
@@ -0,0 +1,44 @@
+type t;
+
+type revisionInfo = {
+  .
+  "revision": string,
+  "folderPath": string,
+  "executablePath": string,
+  "url": string,
+  "local": Js.boolean,
+};
+
+external empty : unit => t = "%identity";
+
+[@bs.send]
+external canDownload : (t, string) => Js.Promise.t(Js.boolean) = "";
+
+[@bs.send]
+external download :
+  (t, string, Js.Undefined.t((float, float) => unit)) =>
+  Js.Promise.t(revisionInfo) =
+  "";
+
+let download =
+    (~revision, ~progressCallback=?, t)
+    : Js.Promise.t(revisionInfo) =>
+  download(t, revision, progressCallback |> Js.Undefined.fromOption);
+
+[@bs.send] external localRevisions : t => Js.Promise.t(array(string)) = "";
+
+[@bs.deriving jsConverter]
+type platform = [
+  | [@bs.as "mac"] `Mac
+  | [@bs.as "linux"] `Linux
+  | [@bs.as "win32"] `Win32
+  | [@bs.as "Win64"] `Win64
```

Comment:
win64 should be all lower case. If you just make the polymorphic variants all lower case and the same as the string doesn't it convert them to the right string? No need to use bs.as.

Reply:
I thought variant constructor need to be capitalized. tried lower case but without success.

Reply:
Polymorphic variants don't need to be capitalized. For example: https://github.com/bs-puppeteer/bs-puppeteer/blob/master/src/ConsoleMessage.re

Reply:
Oh! That works, Thanks!